src/main/java/repositories/impl/map/InventoryItemRepositoryImpl.java

```diff
@@ -39,4 +40,15 @@ public InventoryItem find(Long id) {
     public List<InventoryItem> findAll() {
         return inventoryItemMap.values().stream().collect(Collectors.toList());
     }
+
+    @Override
+    public boolean checkItemAvailability(Item item) {
+        Optional<InventoryItem> inventoryItem = inventoryItemMap
+                .values()
+                .stream()
+                .filter(o -> o.getItem().getId() == item.getId())
+                .findFirst();
+
+        return inventoryItem.get().getQuantity() > 0;
```

Comment:
don't you need to check against how much of the item is required for the dish?
for example:
- Dish 1:
   - Ingredients: 
     - Bacon: 10

but I have only 5 bacons in my inventory, It will return true since there is more than 0 bacon, but I will not be able to complete my dish